plugins/notifiers/lark/client.go

```diff
@@ -0,0 +1,280 @@
+package lark
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"html/template"
+	"net/http"
+
+	"github.com/goto/guardian/pkg/evaluator"
+	"github.com/goto/guardian/pkg/log"
+
+	"github.com/goto/guardian/utils"
+
+	"github.com/goto/guardian/domain"
+)
+
+const (
+	larkHost = "https://open.larksuite.com"
+)
+
+type user struct {
+	ID       string `json:"id"`
+	TeamID   string `json:"team_id"`
+	Name     string `json:"name"`
+	RealName string `json:"real_name"`
+}
+
+type tokenResponse struct {
+	OK     string `json:"msg"`
+	Token  string `json:"tenant_access_token"`
+	Code   int    `json:"code"`
+	Expire int    `json:"expire"`
+}
+
+type Payload struct {
+	AppID     string `json:"app_id"`
+	AppSecret string `json:"app_secret"`
+}
+
+type WorkSpaceConfig struct {
+	Workspaces []LarkWorkspace `mapstructure:"workspaces"`
+}
+
+type LarkWorkspace struct {
+	WorkspaceName string `mapstructure:"workspace" validate:"required"`
+	ClientId      string `mapstructure:"client_id" validate:"required"`
+	ClientSecret  string `mapstructure:"client_secret" validate:"required"`
+	Criteria      string `mapstructure:"criteria" validate:"required"`
+}
+
+type Notifier struct {
+	workspaces []LarkWorkspace
```

Comment:
does lark have concept of workspaces as well? and since we have handled the multitenancy through `Notifiers` I believe we will no longer need this.

Reply:
hmm makes sense.